### PR TITLE
Expand wrroc with fields needed for federated analysis

### PR DESCRIPTION
**Pull request by famosab**
Pull request to reference what we extended.

### CODE REVIEW

File: nextflow.config [hunk 1]
```diff
@@ -23,6 +23,12 @@ prov {
         wrroc {
             file = "${params.outdir}/ro-crate-metadata.json"
             overwrite = true
+            agent {
+                name = "John Doe"
+                orcid = "https://orcid.org/0000-0000-0000-0000"
+            }
```
Comment (ewels) [inline nextflow.config:29]: I think we can pull this information from the new `manifest.contributors` config scope now:

https://www.nextflow.io/docs/latest/reference/config.html#config-manifest

File: nextflow.config [hunk 2]
```diff
@@ -23,6 +23,12 @@ prov {
...
+            license = "https://spdx.org/licenses/Apache-2.0"
```
Comment (ewels) [inline nextflow.config:30]: Can we not pull this information automatically from a `LICEN[CS]E.(md|txt)` file?

File: nextflow.config (hunk 1 above)
Comment (fbartusch) [reply nextflow.config:29]: This new output format for the nf-prov plugin creates RO-Crates that follow [this profile](https://www.researchobject.org/workflow-run-crate/profiles/provenance_run_crate/).
The agent mentioned in the config is not the persons who wrote the workflow, but the person who runs the workflow.

File: nextflow.config (hunk 2 above)
Comment (fbartusch) [reply nextflow.config:30]: This new output format for the nf-prov plugin creates RO-Crates that follow [this profile](https://www.researchobject.org/workflow-run-crate/profiles/provenance_run_crate/).

Similar to your previous review concerning the agent, the license is added directly to the RO-Crate. The crate will include input and output files, the pipeline configuration, and the pipeline itself. I will ensure that the `ro-crate-metadata.json` annotates all Workflow components (i.e., everything created by the original workflow author) with the license specified in a `LICENSE.(md|txt)` file.

File: plugins/nf-prov/src/main/nextflow/prov/WrrocRenderer.groovy [hunk 3]
```diff
@@ -49,213 +69,656 @@ class WrrocRenderer implements Renderer {
...
+
+                def createSoftwareFinal = [
+                    "@id"         : toolNameTask,
+                    "@type"       : "SoftwareApplication",
+                    "description" : softwareMap.getAt(toolNameTask).toString()
+                ]
+                return createSoftwareFinal
+            }
+
+        final wfToolDescriptions = perTool.collect()
```
Comment (famosab) [inline plugins/nf-prov/src/main/nextflow/prov/WrrocRenderer.groovy:428]: @bentsherman This is the new function I wrote. As you can see I did a lot of weird stuff to be able to access the nested yaml file.

File: plugins/nf-prov/src/main/nextflow/prov/WrrocRenderer.groovy [hunk 4]
```diff
@@ -49,213 +69,656 @@ class WrrocRenderer implements Renderer {
...
+            // Construct the path to the meta.yml file
+            Path metaFile = Paths.get(projectDir, 'modules', 'nf-core', toolName, 'meta.yml')
+
+            if (Files.exists(metaFile)) {
+                Yaml yaml = new Yaml()
+                return yaml.load(metaFile.text) as Map
+                }
+            }
+        return null
+        }
```
Comment (famosab) [inline plugins/nf-prov/src/main/nextflow/prov/WrrocRenderer.groovy:719]: @bentsherman Maybe the problem is this function. Ideally I would like to read the meta.yaml into different objects divided by the toplevel key (so name, description, tools, input, output). And hand this with the process to the functions in this groovy class. But maybe there is another solution, too.

**Review by famosab — COMMENTED** [review]
@bentsherman I tried to point out the two main things I worked on :)

File: plugins/nf-prov/src/main/nextflow/prov/WrrocRenderer.groovy (hunk 4 above)
Comment (bentsherman) [reply plugins/nf-prov/src/main/nextflow/prov/WrrocRenderer.groovy:719]: I guess you have to do a bit of guessing to figure out where the `meta.yml` is relative to the main script.

Actually you should be able to get it this way:
```groovy
ScriptMeta.get(processor.getOwnerScript()).getModuleDir().resolve('meta.yml')
```

Comment (bentsherman) [reply plugins/nf-prov/src/main/nextflow/prov/WrrocRenderer.groovy:719]: To use ScriptMeta:
```groovy
import nextflow.script.ScriptMeta
```

**Pull request closed without merging**